### ISSUE TITLE
read 12/24 hour option from locale

### ISSUE THREAD

**Issue by MrYakobo** [insanum/gcalcli#417]
Specifying `--locale=sv_SE.utf8` outputs `6:00` instead of the default `6:00 pm`. I expected `18:00`.

A `--24-hour` option or perhaps parsing the locale to determine 24-hourness would be a nice feature :smile:

**Comment by jcrowgey** [comment]
FWIW, the option `--military` outputs with 24 hour time.  I support the idea of adding `--24-hour` alongside it as an alias because many non-militaristic scenarios call for 24 hour time.

I'm going to update your issue title to emphasize the idea of reading 12/24 hour configuration from locale.

**Comment by yermulnik** [comment]
How do I force `gcalcli` to honor `--military` option?
```
> gcalcli remind --military 360
usage: gcalcli [-h] [--client-id CLIENT_ID] [--client-secret CLIENT_SECRET]
               [--config-folder CONFIG_FOLDER] [--noincluderc]
               [--calendar CALENDAR] [--default-calendar DEFAULTCALENDAR]
               [--locale LOCALE] [--refresh] [--nocache] [--conky] [--nocolor]
               [--lineart {fancy,unicode,ascii}]
gcalcli: error: unrecognized arguments: --military

> gcalcli --military remind 360
usage: gcalcli [-h] [--client-id CLIENT_ID] [--client-secret CLIENT_SECRET]
               [--config-folder CONFIG_FOLDER] [--noincluderc]
               [--calendar CALENDAR] [--default-calendar DEFAULTCALENDAR]
               [--locale LOCALE] [--refresh] [--nocache] [--conky] [--nocolor]
               [--lineart {fancy,unicode,ascii}]
gcalcli: error: unrecognized arguments: --military
```

**Comment by Tyilo** [comment]
See #459

**Comment by jcrowgey** [comment]
Thanks @Tyilo !

**Comment by jcrowgey** [comment]
The flip-flop is because I realized upon closer reading of your comment that while we've now implemented properly reading the time format (12/24) from the locale, we actually haven't fixed the first part of your issue, which is the bug that --locale STR flag doesn't work as expected.  I'm going to leave this closed and I've opened #460 to address the remaining issues.
